File: articles/machine-learning-as-algorithmic-inference/prolog/threshold_review.pl
rate(TP, FP, TN, FN, Accuracy, Precision, Recall) :-
    Total is TP + FP + TN + FN,
    Accuracy is (TP + TN) / Total,
    Precision is TP / (TP + FP),
    Recall is TP / (TP + FN).

:- initialization(main).
main :-
    rate(80, 25, 140, 35, Accuracy, Precision, Recall),
    format('accuracy=~6f~n', [Accuracy]),
    format('precision=~6f~n', [Precision]),
    format('recall=~6f~n', [Recall]),
    halt.

File: articles/machine-learning-as-algorithmic-inference/cpp/threshold_review.cpp
#include <iostream>
#include <iomanip>

int main() {
    double tp = 80.0, fp = 25.0, tn = 140.0, fn = 35.0;
    double total = tp + fp + tn + fn;
    std::cout << std::fixed << std::setprecision(6);
    std::cout << "accuracy=" << (tp + tn) / total << "\n";
    std::cout << "precision=" << tp / (tp + fp) << "\n";
    std::cout << "recall=" << tp / (tp + fn) << "\n";
    return 0;
}
